master/buildbot/test/integration/test_mailnotifier.py

```diff
@@ -19,6 +19,10 @@
 import base64
 
 from twisted.internet import defer
+try:
+    from twisted.mail.smtp import ESMTPSenderFactory
+except ImportError:
+    ESMTPSenderFactory = None
 
 from buildbot.reporters.mail import MailNotifier
```

Comment:
There is a isort issue. Note that same trick is in mail.py, so you can also do:

     from buildbot.reporters.mail import ESMTPSenderFactory

which will be None if twisted mail does an import error